src/WebJobs.Script/Description/FunctionInvokerBase.cs

```diff
@@ -297,4 +204,55 @@ public void Dispose()
             GC.SuppressFinalize(this);
         }
     }
+
+    // Per function instance
+    public class PerInstanceState
```

Comment:
I think you should rename this to FunctionInstanceMonitor since it is more behavioral (Start/Stop) rather than just a state bag